File: tournaments/divisorsPairs/divisorsPairs.cs
int divisorsPairs(int[] sequence) {
    int result = 0;
    
    for (int i = 0; sequence.Length > i; ++i) {
        for (int j = i + 1; sequence.Length > j; ++j) {
            if (0 == sequence[i] % sequence[j] || 0 == sequence[j] % sequence[i]) {
                ++result;
            }
        }
    }
    return result;   
}

File: tournaments/divisorsPairs/divisorsPairs.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub divisorsPairs {
    my ($sequence) = @_;

    my $len = @$sequence;
    my $result = 0;

    for (my $i = 0; $len > $i; ++$i) {
        for (my $j = $i + 1; $len > $j; ++$j) {
            unless ((@$sequence[$i] % @$sequence[$j]) && (@$sequence[$j] % @$sequence[$i])) {
                ++$result;
            }
        }
    }

    return $result;
}
